python/ray/tests/test_generators.py

```diff
@@ -183,15 +183,36 @@ def generator(num_returns, store_in_plasma, counter):
         assert ray.get(ref)[0] == i
 
 
+@pytest.mark.parametrize("use_actors", [False, True])
 @pytest.mark.parametrize("store_in_plasma", [False, True])
-def test_dynamic_generator(ray_start_regular, store_in_plasma):
-    @ray.remote(num_returns="dynamic")
-    def dynamic_generator(num_returns, store_in_plasma):
-        for i in range(num_returns):
-            if store_in_plasma:
-                yield np.ones(1_000_000, dtype=np.int8) * i
-            else:
-                yield [i]
+def test_dynamic_generator(ray_start_regular, use_actors, store_in_plasma):
```

Comment:
shall we also test failure case, similar to `CustomException` above?

Reply:
Sure, will add!

Reply:
Hmm appears that `retry_exception` option isn't supported for actors right now, but I added a test for normal exception handling.

Reply:
use_actors seems to kick in the branch that doesn't use actors, and vice versa